packages/api/AlvTime.Persistence/Repositories/InvoiceRateRepository.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using AlvTime.Business.InvoiceRate;
+using AlvTime.Persistence.DatabaseModels;
+using Microsoft.EntityFrameworkCore;
+
+namespace AlvTime.Persistence.Repositories;
+
+public class InvoiceRateRepository : IInvoiceRateRepository
+{
+
+    private readonly AlvTime_dbContext _context;
+
+    public InvoiceRateRepository(AlvTime_dbContext context)
+    {
+        _context = context;
+    }
+
+    public Task<List<InvoiceRateResponseDto>> GetInvoiceRateStatistics(int userId, DateTime fromDate, DateTime toDate)
+    {
+        return (from hour in _context.Hours
+                join task in _context.Task on hour.TaskId equals task.Id
+                join project in _context.Project on task.Project equals project.Id
+                join customer in _context.Customer on project.Customer equals customer.Id
+                where hour.User == userId
+                      && hour.Date >= fromDate
+                      && hour.Date <= toDate
+                select new InvoiceRateResponseDto
```

Comment:
Kanskje kalle denne noe annet enn InvoiceRateResponseDto siden den også ikke direkte har noe å gjøre med InvoiceRate

Reply:
Det er jo forsåvidt sant. Kalte den for `TimeEntryWithCustomerDto` og flyttet den til ny lokasjon